### PR TITLE
feat: read metadata from write cache

### PR DESCRIPTION
**Pull request by QuenKar**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

- read parquet metadata from write cache.
- refactor some test code.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.
- [x]  This PR does not require documentation updates.

## Refer to a related PR or issue link (optional)
#2965

### CODE REVIEW

File: src/mito2/src/cache.rs [hunk 1]
```diff
@@ -69,15 +70,35 @@ impl CacheManager {
...
+        if metadata.is_some() {
+            return metadata;
+        }
+
+        // Try to get metadata from write cache
+        let key = IndexKey::new(region_id, file_id, FileType::Parquet);
+        if let Some(write_cache) = &self.write_cache {
+            if let Some(metadata) = write_cache
+                .file_cache()
+                .try_get_parquet_meta_data(key, file_size)
```
Comment (evenyag) [inline src/mito2/src/cache.rs:94]: The `IndexValue` in the file cache already has a `file_size` field.

File: src/mito2/src/cache.rs [hunk 2]
```diff
@@ -69,15 +70,30 @@ impl CacheManager {
...
+        if metadata.is_some() {
+            return metadata;
+        }
+
+        // Try to get metadata from write cache
+        let key = IndexKey::new(region_id, file_id, FileType::Parquet);
+        if let Some(write_cache) = &self.write_cache {
+            if let Some(metadata) = write_cache.file_cache().get_parquet_meta_data(key).await {
+                return Some(Arc::new(metadata));
+            }
```
Comment (zhongzc) [inline src/mito2/src/cache.rs:93]: Will anyone put the metadata obtained from the `write_cache` back to the `sst_meta_cache`?

Comment (QuenKar) [reply src/mito2/src/cache.rs:93]: good catch!

**Review by zhongzc — APPROVED** [review]

**Review by evenyag — APPROVED** [review]
LGTM

**Pull request merged**